Add link to Privacy Policy on Login and Registration screens

mcliwanow opened the pull request:
#### Proposed Changes

There was a request  p6JqRr-6Yz-p2#comment-14399 to add text “and acknowledge that you have read our Privacy Policy” to the TOS acceptance lines for both Login and Registration screens.

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- On WooCommerce.test click login
- Replace _https://wordpress.com/_ with _http://calypso.localhost:3000/_ in the URL and check if texts mentioning Terms of Service contain a link to Privacy Policy too
- Go to the registration screen (click "Create an Account" link) and check if texts mentioning Terms of Service contain a link to Privacy Policy too
- Go to the regular login screen http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fwordpress.com%2F and check if texts mentioning Terms of Service contain a link to Privacy Policy too
- Click on the "Create a new account" link and check if texts mentioning Terms of Service contain a link to Privacy Policy too
- Please test around this issue

#### Screenshots

![image](https://user-images.githubusercontent.com/4765119/182578595-3341dc6e-8e55-477c-9152-4cf3fb08f6e9.png)

![image](https://user-images.githubusercontent.com/4765119/182578677-c164925e-8790-4aa6-b619-24cf5945da70.png)

http://calypso.localhost:3000/log-in/new/en-gb?redirect_to=http%3A%2F%2Fcalypso.localhost%3A3000%2Fstart%2Fdomains%3Fuser_email%3D&signup_url=%2Fstart%2Fuser%3Fuser_email%3D
![image](https://user-images.githubusercontent.com/4765119/182578743-f17e0ec6-fe13-4cf7-8289-d11124ddc4a6.png)

http://calypso.localhost:3000/start/user?user_email=
![image](https://user-images.githubusercontent.com/4765119/182578823-32e5298a-26d8-4e3c-ae58-7c942709c7cc.png)

http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fwordpress.com%2F
![image](https://user-images.githubusercontent.com/4765119/182578879-96466919-681f-45b9-ab1c-d466e3815886.png)

mcliwanow committed:
Add link to Privacy Policy on Login and Registration screens

There was a request to add text “and acknowledge that you have read our Privacy Policy” to the TOS acceptance lines for both Login and Registration screens.

tyxla commented on client/blocks/login/login-form.jsx:
```diff
@@ -598,7 +598,9 @@ export class LoginForm extends Component {
-							'By continuing, ' + 'you agree to our {{tosLink}}Terms of Service{{/tosLink}}.',
+							'By continuing, ' +
+								'you agree to our {{tosLink}}Terms of Service{{/tosLink}} and' +
+								' acknowledge that you have read our {{privacyLink}}Privacy Policy{{/privacyLink}}.',
```
Let's make sure to get a confirmation from legal folks that the copy here is right - perhaps @melanieharari or @catscheer? This is in compliance with the comment above FWIW.

mcliwanow replied:
@tyxla  I see that the link I initially used in the issue description wasn't correct 🤦 It's fixed now. It should have been:  p6JqRr-6Yz-p2#comment-14399 - you can see there that the change was accepted by the legal team.

tyxla replied:
Perfect, thanks for confirming @mcliwanow 👍

melanieharari replied:
Copy looks good. Thanks for the ping!

tyxla commented on client/blocks/login/social.jsx:
```diff
@@ -157,17 +157,29 @@ class SocialLoginForm extends Component {
-							},
-						} ) }
+						{ translate(
+							'By continuing, you agree to our {{tosLink}}Terms of' +
```
Same here for legal confirmation.

melanieharari replied:
Copy here also looks good, thanks!

tyxla commented on client/blocks/login/social.jsx:
```diff
@@ -182,17 +194,25 @@ class SocialLoginForm extends Component {
 									rel="noopener noreferrer"
 								/>
 							),
+							privacyLink: (
```
I realize some of that repetition was here before, but it's a bummer to repeat those a few times in the same component. Could we try to reuse them?

tyxla replied:
Feel free to follow up with that in another PR if that makes it easier FWIW.

mcliwanow replied:
Thanks! It's a good suggestion 👍 Changed in https://github.com/Automattic/wp-calypso/pull/66220/commits/b0bb2d9b6e39bf15981db9cdc201db9417fd07ce

tyxla commented on client/blocks/signup-form/index.jsx:
```diff
@@ -783,43 +783,36 @@ class SignupForm extends Component {
-	getTermsOfServiceUrl() {
-		return localizeUrl( 'https://wordpress.com/tos/' );
-	}
+	handleOnClickPrivacy = () => {
```
Naming nit: `onClick` usually points to a function that will be executed on click. `handleClick` is usually that function. `handleOnClick` is a mix of both that I don't think makes much sense. Perhaps we should change this to just `handlePrivacyClick`?

mcliwanow replied:
Sure, I'll change it. The convention was copied from the function above (`handleOnClickTos`) but then I'll update both of them.

tyxla replied:
Thanks for that!

mcliwanow replied:
Function names updated in https://github.com/Automattic/wp-calypso/pull/66220/commits/b0bb2d9b6e39bf15981db9cdc201db9417fd07ce

tyxla reviewed (commented):
This generally looks good from a frameworks perspective. The only reservation I have is the code repetition, which seems to grow with the newly introduced privacy link. Could we consider reusing those same components throughout the codebase?

Also, let's make sure to get a confirmation from the legal folks that the copy updates make sense from their perspective.

tyxla commented:
I have no other feedback here, this looks good from my perspective and I'll let y'all test and review it within your team 😉 Thanks, and let me know if you need anything in the meantime.

mcliwanow committed:
Update names of click handlers, extract duplicated links to variables

After code review feedback, names of the functions that handle clicks on ToS and Privacy Policy were updated. We also had a duplicated links when rendering social ToS - those were extracted to a separate variables and reused later.

andfinally commented on client/blocks/login/social.jsx:
```diff
@@ -153,21 +153,37 @@ class SocialLoginForm extends Component {
 			redirectTo.includes( 'jetpack/connect' ) &&
 			config.isEnabled( 'jetpack/magic-link-signup' );
+
+		const privacyLink = (
+			<a
+				href={ localizeUrl( 'https://automattic.com/privacy/' ) }
+				target="_blank"
+				rel="noopener noreferrer"
+			/>
+		);
+		const tosLink = (
+			<a
+				href={ localizeUrl( 'https://wordpress.com/tos/' ) }
+				target="_blank"
+				rel="noopener noreferrer"
+			/>
+		);
```
We seem to do things in the order `tosLink`, `privacyLink` elsewhere, so this upsets my Obsessive Compulsive Disorder, but I'll live.

mcliwanow replied:
@andfinally , I changed the order in https://github.com/Automattic/wp-calypso/pull/66220/commits/55dc9c022dec8a3ff001ef4d9504508c8bd229bb so you won't have to worry about it in the future 😉

andfinally reviewed (approved):
Looks good, works as expected, thanks! 👌

KokkieH reviewed (approved):
Everything works as described for me.

mcliwanow committed:
Change order of ToS and Privacy links (for consistency).

Also, small update to margin of login form terms (for Woo login).

mcliwanow committed:
Fix input styles on the Login page in development mode

In development mode, input styles `.login__form .login__form-userdata input` and `input[type=text].form-text-input` have the same specifity, which means that whatever is loaded last is applied. In case of the login page, in development mode, `.login__form .login__form-userdata input` is loaded first. On production `.login__form .login__form-userdata input` is loaded after `input[type=text].form-text-input` so the styles look correctly. This commit ensures that styles in login/styles.css are more specific than `input[type=text].form-text-input`.

mcliwanow commented:
@tyxla , fyi I did one small change in commit https://github.com/Automattic/wp-calypso/pull/66220/commits/ef7ec27e9fbbd7a7a2f57bbf5b1a5764afa61628 that ensures that input styles on the Login screen during development are properly calculated. 

Without these changes, the selector was just as specific as another selector from "base" styles, and in development mode, the styles were loaded in a different order than in the production build.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7464766

Thank you @mcliwanow for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7464766).
